base/iterators.jl

```diff
@@ -1095,4 +1094,33 @@ eltype(::Type{Stateful{T, VS}} where VS) where {T} = eltype(T)
 IteratorEltype(::Type{Stateful{T,VS}}) where {T,VS} = IteratorEltype(T)
 length(s::Stateful) = length(s.itr) - s.taken
 
+"""
+    only(x)
+
+Returns the one and only element of collection `x`, and throws an error if the collection
+has zero or multiple elements.
+"""
+Base.@propagate_inbounds function only(x)
+    i = iterate(x)
+    @boundscheck if i === nothing
+        error("Collection is empty, must contain exactly 1 element")
+    end
+    (ret, state) = i
+    j = iterate(x, state)
+    @boundscheck if j !== nothing
+        error("Collection has multiple elements, must contain exactly 1 element")
```

Comment:
Shoud we specify how many elements or not? i.e. `"Collection has $(length(x)) elements, must contain exactly 1 element"`

Reply:
No, because it might be infinite, or so large that `collect`ing it uses all the RAM